source/dbo/Stored Procedures/Unmatched Specimens/uspGenerateUnmatchedSpecimens.sql

```diff
@@ -0,0 +1,84 @@
+﻿CREATE PROCEDURE [dbo].[uspGenerateUnmatchedSpecimens]
+AS
+BEGIN TRY
+
+	TRUNCATE TABLE [dbo].[UnmatchedSpecimens]
+
+	SELECT ls.ReferenceLaboratoryNumber 
+	INTO #NoMatchesFound
+		FROM [$(NTBS_Specimen_Matching)].dbo.LabSpecimen ls
+		INNER JOIN vwNotificationYear ny ON ny.NotificationYear = YEAR(ls.SpecimenDate)
+		LEFT JOIN [$(NTBS_Specimen_Matching)].dbo.NotificationSpecimenMatch nsm ON nsm.ReferenceLaboratoryNumber = ls.ReferenceLaboratoryNumber
+		WHERE nsm.MatchType IS NULL
+
+	CREATE TABLE #MatchTypeRanking ([Rank] int, ResultName nvarchar(20) )
+
+	INSERT INTO #MatchTypeRanking VALUES (1, 'Confirmed')
+	INSERT INTO #MatchTypeRanking VALUES (1, 'Possible')
+	INSERT INTO #MatchTypeRanking VALUES (2, 'Rejected-Possible')
+	INSERT INTO #MatchTypeRanking VALUES (2, 'Rejected')
+
+	SELECT ranked.ReferenceLaboratoryNumber INTO #AllMatchesRejected FROM (
+		SELECT DISTINCT nsm.ReferenceLaboratoryNumber, FIRST_VALUE(mtr.ResultName) OVER (PARTITION BY nsm.ReferenceLaboratoryNumber ORDER BY mtr.[rank] ASC) AS [Result]
+			FROM [$(NTBS_Specimen_Matching)].dbo.NotificationSpecimenMatch nsm
+			INNER JOIN [$(NTBS_Specimen_Matching)].dbo.LabSpecimen ls ON ls.ReferenceLaboratoryNumber = nsm.ReferenceLaboratoryNumber
+			INNER JOIN vwNotificationYear ny ON ny.NotificationYear = YEAR(ls.SpecimenDate)
+			INNER JOIN #MatchTypeRanking mtr ON mtr.ResultName = nsm.MatchType
+			) ranked
+		WHERE ranked.Result = 'Rejected' OR ranked.Result = 'Rejected-Possible'
```

Comment:
I think you could use `FIRST_VALUE(mtr.[rank])` in the select and then `WHERE ranked.Rank = 2` here to simplify it a little. 

I think at the moment you could get duplicates here, if there was both a "Rejected" and a "Rejected-Possible" specimen (though I don't know if that's actually possible). But the change suggested above would remove those duplicates if they can occur.